### DISCUSSION TITLE
Testing and benchmarking performance test

### DISCUSSION THREAD

**Discussion by TheGreatRefrigerator** [GIScience/openrouteservice#1986] [Infrastructure]
We should integrate a test suite(s) that checks for the performance of specific request scenarios.
One idea could be to have a slightly bigger test pbf that gets built regularly on master, like once a day/week, and tests the latest code with specific query/response time benchmarks.

**Comment by bujoralexandru** [comment]
We've recently stumbled upon your product and we are really delighted with the way it is going so far in our tests, with some minor exceptions (see https://github.com/GIScience/openrouteservice/issues/1410). 
This being said, we've also started a benchmarking project of our own deployment of your solution (even though our deployment might not be yet optimal for what you're trying to achieve, we run it on a 24-core, 196 GB RAM DigitalOcean memory optimised instance). Another detail is that, for now, we've only build the `driving-car` profile, but we'll surely need the `driving-hgv` profile aswell.

If it is of any help, I can link our GitHub project to this issue when we progress with our findings.

Once again, really nice work with the ORS platform 🎊.

**Comment by MichaelsJP** [comment]
@bujoralexandru Thank you in advance already :). It's always interesting to see how everything performs on different systems.

You are welcome to reach out to us here if you encounter technical bugs or in our [ask forum](https://ask.openrouteservice.org).
